src/components/appointments/types.ts

```diff
@@ -0,0 +1,39 @@
+import type {AppointmentProduct} from '@/data/appointments';
+
+export type AppoinmentStep = 'producten' | 'kalender' | 'contactgegevens';
+
+/**
+ * Information gathered per step to create an appointment.
+ *
+ * The keys are the slugs/identifiers of the individual steps.
+ */
+export interface AppointmentDataByStep {
+  producten: {
+    products: AppointmentProduct[];
+  };
+  kalender: {
```

Comment:
This might go a bit far, but wouldn't it be nice to define the AppointmentSteps ('product', 'kalender', 'contactgegevens') as const/enum, and use their values in the `AppointmentDataByStep` type definition?

Just to make it super obvious that these values should not be changed. We could then also re-use them, instead of using/relaying on *"magic"* strings.

Something like:
```suggestion
enum AppointmentSteps {
  PRODUCTEN = 'producten',
  KALENDER = 'kalender',
  CONTACTGEGEVENS = 'contactgegevens',
}

/**
 * Information gathered per step to create an appointment.
 *
 * The keys are the slugs/identifiers of the individual steps.
 */
export interface AppointmentDataByStep {
  [AppointmentSteps.PRODUCTEN]: {
    products: AppointmentProduct[];
  };
  [AppointmentSteps.KALENDER]: {
  ...
```

Reply:
Yeah your setup defines `AppointmentDataByStep` clearly, and with `AppoinmentStep` you can target the individual steps clearly. But this all relays on manually making sure the strings are still aligned.

I agree that:
```
export interface AppointmentDataByStep {
  producten: {
    products: AppointmentProduct[];
  };
}
```
is far easier to read, but it also feels kinda loose/"fragile". Definitely as this is suppose to connect to Formio keys, and any typo could cause issues.

Reply:
> Definitely as this is suppose to connect to Formio keys

that's not true though, the keys are fixed and well defined, only the `contactgegevens` step has a nested object with Formio keys inside and that's typed as `JSONObject`

The majority of the appointment flow does *not* use Formio.